runtime/runtime_test.go

```diff
@@ -369,6 +369,70 @@ func TestServerInitialized(t *testing.T) {
 	}
 }
 
+func TestWarningMessage(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Millisecond)
+	defer cancel() // NOTE(sr): The timeout will have been reached by the time `done` is closed.
+
+	params := NewParams()
+
+	logger := logging.New()
+	stdout := bytes.NewBuffer(nil)
+	logger.SetOutput(stdout)
+
+	params.Logger = logger
+	params.Addrs = &[]string{":8181"}
+	params.GracefulShutdownPeriod = 1
+	rt, err := NewRuntime(ctx, params)
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+
+	done := make(chan struct{})
+	go func() {
+		rt.StartServer(ctx)
+		close(done)
+	}()
+	<-done
+
+	expected := "You are not binding to localhost. This could have security implications if not intended. Please see https://www.openpolicyagent.org/docs/latest/security/#interface-binding for more information."
+	if !strings.Contains(stdout.String(), expected) {
+		t.Fatalf("Expected output to contain: \"%v\" but got \"%v\"", expected, stdout.String())
+	}
+
+}
+
+func TestNoWarningMessage(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Millisecond)
+	defer cancel() // NOTE(sr): The timeout will have been reached by the time `done` is closed.
+
+	params := NewParams()
+
+	logger := logging.New()
+	stdout := bytes.NewBuffer(nil)
+	logger.SetOutput(stdout)
+
+	params.Logger = logger
+	params.Addrs = &[]string{"localhost:8181"}
+	params.GracefulShutdownPeriod = 1
+	rt, err := NewRuntime(ctx, params)
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+
+	done := make(chan struct{})
+	go func() {
+		rt.StartServer(ctx)
+		close(done)
+	}()
+	<-done
+
+	expected := ""
+	if !strings.Contains(stdout.String(), expected) {
+		t.Fatalf("Expected output to contain: \"%v\" but got \"%v\"", expected, stdout.String())
```

Comment:
"Expected no output" might be a better description here.

Reply:
Doh, yeah that's more sensical :sweat_smile: will update :+1:

Reply:
:) Should perhaps be an equals check rather than contains too?

Reply:
Hey @anderseknert! :wave: 

I need to check a bit more but I am not sure I can do `equals` because this is the output you get from the logger in the tests:

```
"time="2023-01-27T15:59:13+01:00" level=info msg="Initializing server." addrs="[localhost:8181]" diagnostic-addrs="[]"
        time="2023-01-27T15:59:13+01:00" level=info msg="Shutting down..."
        time="2023-01-27T15:59:13+01:00" level=info msg="Server shutdown."
```

I am not 100% sure how I would try to target it specifically, I am thinking something a long the lines of:

```go
	unexpectedWarning := "OPA running on a public (0.0.0.0) network interface. Unless you intend to expose OPA outside of the host, binding to the localhost interface (--addr localhost:8181) is recommended. See https://www.openpolicyagent.org/docs/latest/security/#interface-binding"
	if strings.Contains(stdout.String(), unexpectedWarning) {
		t.Fatalf("Expected output to contain: \"%v\" but got \"%v\"", unexpectedWarning, stdout.String())
	}
```

But I will do a bit more research to see if I can make it a bit more nice :smile:

Reply:
Oh, that's not important. If what you have works, go with that.